containers/DvmState.ts

```diff
@@ -0,0 +1,140 @@
+import EmpState from "./EmpState";
+
+const { formatUnits: fromWei, parseBytes32String: hexToUtf8 } = utils;
+
+interface ContractState {
+  hasEmpPrice: boolean | null;
+  resolvedPrice: number | null;
+}
+
+const initState = {
+  hasEmpPrice: null,
+  resolvedPrice: null,
+};
+
+// Taken from Voting.sol implementation.
+enum PRICE_REQUEST_STATUSES {
+  NOT_REQUESTED, // Was never requested.
+  ACTIVE, // Is being voted on in the current round.
+  RESOLVED, // Was resolved in a previous round.
+  FUTURE, // Is scheduled to be voted on in a future round.
+}
+
+interface PriceRequestQuery {
+  identifier: {
+    id: string;
+  };
+  time: string;
+}
+
+const useContractState = () => {
+  const { block$ } = Connection.useContainer();
+  const { contract: dvm } = DvmContract.useContainer();
+  const { empState } = EmpState.useContainer();
+  const { priceIdentifier, expirationTimestamp } = empState;
+
+  const [state, setState] = useState<ContractState>(initState);
+
+  // Because apollo caches results of queries, we will poll/refresh this query periodically.
+  // We set the poll interval to a very slow 5 seconds for now since the position states
+  // are not expected to change much.
+  // Source: https://www.apollographql.com/docs/react/data/queries/#polling
+  const { loading, error, data } = useQuery(PRICE_REQUESTS, {
+    context: { clientName: "UMA" },
+    pollInterval: 5000,
+  });
+
+  // get state from EMP
+  const queryState = async () => {
+    if (
+      dvm !== null &&
+      priceIdentifier !== null &&
+      expirationTimestamp !== null
+    ) {
+      const res = await Promise.all([
+        dvm.getPriceRequestStatuses([
+          {
+            identifier: priceIdentifier.toString(),
+            time: expirationTimestamp.toNumber(),
+          },
+        ]),
+      ]);
+
+      const hasPrice = res[0][0].status === PRICE_REQUEST_STATUSES.RESOLVED;
+
+      let resolvedPrice = null;
+
+      // Fetch price from graph if available.
+      if (error) {
+        console.error(`Apollo client failed to fetch graph data:`, error);
+      } else if (hasPrice && !loading && data) {
+        // Implementation 1: Use subgraph data. This works for mainnet but not Kovan since the UMA subgraph does not currently index Kovan data.
+        const matchingPriceRequest = data.priceRequests.find(
+          (request: PriceRequestQuery) => {
+            return (
+              request.identifier.id === hexToUtf8(priceIdentifier) &&
+              request.time === expirationTimestamp.toString()
+            );
+          }
+        );
+        if (matchingPriceRequest && matchingPriceRequest.isResolved) {
+          resolvedPrice = parseFloat(
+            fromWei(matchingPriceRequest.price.toString())
+          );
+        }
+
+        // // Implementation 2: Call contract. This works for Kovan/MockOracle and is therefore useful for testing, but not mainnet since `getPrice` must be called with the overrideL `{from: emp.address}`.
```

Comment:
I figured out how to do it! WHen I construct the `dvm` contract, instead of passing in the `signer` I can just pass in the `provider`. Without a default signer, you can call a contract and override the `from`!